packages/devtools_app/lib/src/memory/memory_controller.dart

```diff
@@ -48,34 +51,63 @@ class MemoryController {
 
   int selectedSample = -1;
 
-  bool _paused = false;
+  static const String liveFeed = 'Live Feed';
 
-  bool get paused => _paused;
+  /// Notifies that the source of the memory feed has changed.
+  ValueListenable get memorySourceNotifier => _memorySourceNotifier;
 
-  void pauseLiveFeed() {
-    _paused = true;
+  final _memorySourceNotifier = ValueNotifier<String>(liveFeed);
+
+  set memorySource(String source) {
+    _memorySourceNotifier.value = source;
   }
 
-  void resumeLiveFeed() {
-    _paused = false;
+  String get memorySource => _memorySourceNotifier.value;
+
+  /// MemorySource has changed update the view.
+  void updatedMemorySource() {
```

Comment:
instead of doing all this in a separate method, execute this code immediately after memorySource changes (in the selectMemorySource method)